types/dom-mediacapture-record/index.d.ts

```diff
@@ -0,0 +1,64 @@
+// Type definitions for w3c MediaStream Recording 1.0
+// Project: https://w3c.github.io/mediacapture-record
+// Definitions by: Elias Meire <https://github.com/eliasmeire>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+interface MediaRecorderErrorEventInit extends EventInit {
+    error: DOMException;
+}
+
+declare class MediaRecorderErrorEvent extends Event {
+    constructor(type: string, eventInitDict: MediaRecorderErrorEventInit);
+    readonly error: DOMException;
+}
+
+interface BlobEventInit extends EventInit {
+    data: Blob;
+    timecode?: number;
+}
+
+declare class BlobEvent extends Event {
+    constructor(type: string, eventInitDict: BlobEventInit);
+    readonly data: Blob;
+    readonly timecode: number;
+}
+
+interface MediaRecorderOptions {
+    mimeType?: string;
+    audioBitsPerSecond?: number;
+    videoBitsPerSecond?: number;
+    bitsPerSecond?: number;
+}
+
+type RecordingState = 'inactive' | 'recording' | 'paused';
+
+declare class MediaRecorder extends EventTarget {
+    readonly stream: MediaStream;
+    readonly mimeType: string;
+    readonly state: RecordingState;
+    readonly videoBitsPerSecond: number;
+    readonly audioBitsPerSecond: number;
+
+    ondataavailable: (event: BlobEvent) => void;
```

Comment:
With `"strictFunctionTypes": true` this throws an error:

https://travis-ci.org/DefinitelyTyped/DefinitelyTyped/builds/455512797
```
ERROR: 43:44  expect  TypeScript@next compile error: 
Argument of type '(event: BlobEvent) => void' is not assignable to parameter of type 'EventListener | EventListenerObject | null'.
  Type '(event: BlobEvent) => void' is not assignable to type 'EventListener'.
    Types of parameters 'event' and 'evt' are incompatible.
      Type 'Event' is missing the following properties from type 'BlobEvent': data, timecode
```

but I would like to have type hinting for BlobEvent :)